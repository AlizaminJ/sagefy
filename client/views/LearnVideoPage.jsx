/* eslint-disable camelcase */

import React from 'react'
import { string, shape, number } from 'prop-types'
import { convertUuidToUuid58 as to58 } from 'uuid58'
import Layout from './components/Layout'
import Icon from './components/Icon'

function getVideoUrl(site, video_id) {
  if (site === 'youtube')
    return `https://www.youtube.com/embed/${video_id}?autoplay=1&amp;modestbranding=1&amp;rel=0`
  if (site === 'vimeo') return `https://player.vimeo.com/video/${video_id}`
  return 'https://example.com'
}

export default function LearnVideoPage({
  hash,
  progress,
  card: {
    entityId,
    name,
    data: { site, video_id },
  },
}) {
  return (
    <Layout
      hash={hash}
      page="LearnVideoPage"
      title="Learn"
      description="-"
      canonical={`/video-cards/${to58(entityId)}`}
    >
      {progress && (
        <section>
          <progress value={progress} />
        </section>
      )}
      <section>
        <iframe
          src={getVideoUrl(site, video_id)}
          width="600"
          height="400"
          allowFullScreen="true"
          title={name}
        />
      </section>
      <section>
        <form action="/next">
          <button type="submit">
            <Icon i="card" /> Next Card
          </button>
        </form>
      </section>
    </Layout>
  )
}

LearnVideoPage.propTypes = {
  hash: string.isRequired,
  progress: number,
  card: shape({
    name: string.isRequired,
    data: shape({
      video_id: string.isRequired,
    }).isRequired,
  }).isRequired,
}

LearnVideoPage.defaultProps = {
  progress: null,
}
