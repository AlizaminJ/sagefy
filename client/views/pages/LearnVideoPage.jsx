/* eslint-disable camelcase */

import React from 'react'
import { string, shape, number } from 'prop-types'
import Icon from '../components/Icon'

export default function LearnVideoPage({ progress, name, data: { video_id } }) {
  return (
    <div className="LearnVideoPage">
      {progress && (
        <section>
          <progress value={progress} />
        </section>
      )}

      <section>
        <iframe
          src={`https://www.youtube.com/embed/${video_id}?autoplay=1&amp;modestbranding=1&amp;rel=0`}
          width="600"
          height="400"
          allowFullScreen="true"
          title={name}
        />
      </section>

      <section>
        <form action="/next">
          <button type="submit">
            <Icon i="card" /> Next Card
          </button>
        </form>
      </section>
    </div>
  )
}

LearnVideoPage.propTypes = {
  progress: number,
  name: string.isRequired,
  data: shape({
    video_id: string.isRequired,
  }).isRequired,
}

LearnVideoPage.defaultProps = {
  progress: null,
}
