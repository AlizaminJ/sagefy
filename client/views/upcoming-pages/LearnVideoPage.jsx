import React from 'react'
import Icon from '../components/Icon'
import Meter from '../components/Meter'

export default function LearnVideoPage() {
  return (
    <div className="LearnVideoPage">
      <section>
        <Meter width={0.7} />
      </section>

      <section>
        <iframe
          className="video lift"
          src="https://www.youtube.com/embed/qAojF80nwsM?autoplay=1&amp;modestbranding=1&amp;rel=0"
          width="600"
          height="400"
          allowFullScreen="true"
          frameBorder="0"
          title="Learn Video"
        />
      </section>

      <section>
        <form action="/learn-choice/1">
          <button type="submit">
            <Icon i="card" /> Next Card
          </button>
        </form>
      </section>
    </div>
  )
}
