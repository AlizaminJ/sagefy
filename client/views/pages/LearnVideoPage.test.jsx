import LearnVideoPage from './LearnVideoPage'

describe('LearnVideoPage', () => {
  it('should render LearnVideoPage', () => {
    expect(LearnVideoPage({ card: { data: {} } })).toMatchSnapshot()
  })
})
