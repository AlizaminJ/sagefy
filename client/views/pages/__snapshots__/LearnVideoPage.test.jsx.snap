// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnVideoPage should render LearnVideoPage 1`] = `
<Layout
  description="-"
  page="LearnVideoPage"
  title="Learn"
>
  <section>
    <iframe
      allowFullScreen="true"
      height="400"
      src="https://example.com"
      width="600"
    />
  </section>
  <section>
    <form
      action="/next"
    >
      <button
        type="submit"
      >
        <Icon
          i="card"
          s="m"
        />
         Next Card
      </button>
    </form>
  </section>
</Layout>
`;
